src/engine/GroupBy.cpp

```diff
@@ -717,23 +708,24 @@ bool GroupBy::computeGroupByForJoinWithFullScan(IdTable* result) {
     currentCount += currentCardinality;
   }
   pushRow();
-  *result = std::move(idTable).toDynamic();
-  return true;
+  return std::move(idTable).toDynamic();
 }
 
 // _____________________________________________________________________________
-bool GroupBy::computeOptimizedGroupByIfPossible(IdTable* result) {
-  if (computeGroupByForSingleIndexScan(result)) {
-    return true;
-  } else if (computeGroupByForFullIndexScan(result)) {
-    return true;
-  } else if (computeGroupByForJoinWithFullScan(result)) {
-    return true;
-  } else if (computeGroupByObjectWithCount(result)) {
-    return true;
-  } else {
-    return false;
+std::optional<IdTable> GroupBy::computeOptimizedGroupByIfPossible() {
+  if (auto result = computeGroupByForSingleIndexScan()) {
+    return result;
   }
+  if (auto result = computeGroupByForFullIndexScan()) {
+    return result;
+  }
+  if (auto result = computeGroupByForJoinWithFullScan()) {
+    return result;
+  }
+  if (auto result = computeGroupByObjectWithCount()) {
+    return result;
+  }
+  return std::nullopt;
```

Comment:
C++23 makes this easier, by allowing to chain [`or_else`](https://en.cppreference.com/w/cpp/utility/optional/or_else)

Reply:
Then please add the `TODO<C++23>` use std::optional::or_else`.